tests/test_measureCrosstalk.py

```diff
@@ -0,0 +1,128 @@
+            simulated exposures.
+
+        Returns
+        -------
+        coeffErr : `np.ndarray`
+            Array of booleans indicating if the measured and expected
+            crosstalk ratios are smaller than the measured uncertainty
+            in the crosstalk ratio.
+        """
+        mockTask = isrMock.CalibratedRawMock()
+        mockTask.config.rngSeed = 12345
+        mockTask.config.doGenerateImage = True
+        mockTask.config.doAddSky = True
+        mockTask.config.doAddSource = True
+        mockTask.config.doAddCrosstalk = True
+        mockTask.config.doAddBias = True
+        mockTask.config.doAddFringe = False
+
+        mockTask.config.skyLevel = 0.0
+        mockTask.config.biasLevel = 0.0
+        mockTask.config.readNoise = 100.0
+
+        mcConfig = MeasureCrosstalkConfig()
+        mcConfig.extract.threshold = 4000
+        mcConfig.extract.isTrimmed = isTrimmed
+        mct = MeasureCrosstalkTask(config=mcConfig)
+        fullResult = []
+
+        mockTask.config.isTrimmed = isTrimmed
+        # Generate simulated set of exposures.
+        for idx in range(0, 10):
+            mockTask.config.rngSeed = 12345 + idx * 1000
+
+            # Allow each simulated exposure to have nSources random
+            # bright sources.
+            mockTask.config.sourceAmp = (np.random.randint(8, size=nSources)).tolist()
+            mockTask.config.sourceFlux = ((np.random.random(size=nSources) * 25000.0 + 20000.0).tolist())
+            mockTask.config.sourceX = ((np.random.random(size=nSources) * 100.0).tolist())
+            mockTask.config.sourceY = ((np.random.random(size=nSources) * 50.0).tolist())
+
+            exposure = mockTask.run()
+            result = mct.extract.run(exposure)
+            fullResult.append(result.outputRatios)
+
+        # Generate the final measured CT ratios, uncertainties, pixel counts.
+        finalResult = mct.solver.run(fullResult)
+        calib = finalResult.outputCrosstalk
+
+        # Needed because measureCrosstalk cannot find coefficients equal to 0.0
+        coeff = np.nan_to_num(calib.coeffs)
+        coeffSig = np.nan_to_num(calib.coeffErr)
+
+        # Compare result against expectation used to create the simulation.
+        expectation = isrMock.CrosstalkCoeffMock().run()
+        coeffErr = abs(coeff - expectation) <= coeffSig
+        return coeffErr
+
+    def testMeasureCrosstalkTaskTrimmed(self):
+        """Measure crosstalk from a sequence of trimmed mocked images.
+        """
+        coeffErr = self.setup_measureCrosstalk(isTrimmed=True, nSources=8)
+
+        self.assertTrue(np.all(coeffErr))
+
+    def testMeasureCrosstalkTaskUntrimmed(self):
+        """Measure crosstalk from a sequence of untrimmed mocked images.
+        """
+        coeffErr = self.setup_measureCrosstalk(isTrimmed=False, nSources=8)
+
+        # DM-18528 This doesn't always fully converge, so be permissive
+        # for now.
+        self.assertTrue(np.any(coeffErr))
```

Comment:
I've updated the comment to point to the new ticket for this issue: DM-26031.  I think the untrimmed case fails because sources end up outside of the science region.

Reply:
We should discuss that on that ticket, so will mark as resolved, but that does feel like it should be possible (though I've not thought on it much tbh).